### PR TITLE
【CINN】refactor ir_visitor

### PR DESCRIPTION
**Pull request by Courtesy-Xs**
### PR types

Performance optimization
### PR changes

Others
### Description

This PR delete middle ir_visitor class and thus we can avoid middle virtual function call and codes look more clean 
pcard-72718

### CODE REVIEW

File: paddle/cinn/optim/ir_copy.cc [hunk 1]
```diff
@@ -419,19 +421,19 @@ struct IRCopyVisitor : public ir::IRVisitorBase<Expr> {
...
   NODETY_BINARY_OP_FOR_EACH(OP_BINARY_HANDLE)
 #undef OP_BINARY_HANDLE
 
-#define OP_UNARY_HANDLE(op__)                \
-  Expr Visit(const op__* op) override {      \
-    auto v = IRVisitorBase::Visit(&op->v()); \
-    return op__::Make(v);                    \
+#define OP_UNARY_HANDLE(op__)                                \
+  Expr Visit(const op__* op) override {                      \
+    auto v = IRVisitorRequireReImplVisitor::Visit(&op->v()); \
```
Comment (zhhsplendid) [inline paddle/cinn/optim/ir_copy.cc]: Why are 2 Visitors in name?

Using `IrVisitorRequireReImpl`or `IrRequireReImplVisitor`?

File: paddle/cinn/pybind/ir.cc
Comment (zhhsplendid) [inline paddle/cinn/pybind/ir.cc:1]: Why is there an empty file? Should we delete it?

File: paddle/cinn/optim/ir_copy.cc (hunk 1 above)
Comment (Courtesy-Xs) [reply paddle/cinn/optim/ir_copy.cc]: done

File: paddle/cinn/pybind/ir.cc
Comment (Courtesy-Xs) [reply paddle/cinn/pybind/ir.cc:1]: done

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**